eglot.el

```
@@ -2542,27 +2552,32 @@ code actions at point"
                  :context
                  `(:diagnostics
                    [,@(cl-loop for diag in (flymake-diagnostics beg end)
-                               when (cdr (assoc 'eglot-lsp-diag (eglot--diag-data diag)))
-                               collect it)]))))
+                               when (cdr (assoc 'eglot-lsp-diag
+                                                (eglot--diag-data diag)))
+                               collect it)]
+                   ,@(when action-kind
+                       `(:only ,action-kind))))))
          (menu-items
-          (or (mapcar (jsonrpc-lambda (&rest all &key title &allow-other-keys)
-                        (cons title all))
-                      actions)
-              (eglot--error "No code actions here")))
+          (or (cl-loop for action across actions
```

Are we sure `actions` is _always_ an array?  I'm fine if we are, but I've been bitten by this before in Eglot.

---

Well, I've checked `type-of` of the `jsonrpc-request` result and it was `vector`. For me specifying `in` instead `across` gives no results. What is the proper way to loop over actions?

---

If something can be a `vector` or a `list` (which includes `nil`) you need to use an iteration variable and a function which can runtime dispatch on the type of sequences.  But if you know it will be a vector, `across` is fine.  The fact that `in` silently fails in a bug in `cl-loop`, in my opinion.

---

Will using `append` be acceptable here?

```elisp
(cl-loop for action in (append actions nil)
```

---

I use that trick sometimes, yes.  It makes a cons cell for every action, which could be wasteful.  But here it isn't too bad because you're not likely to be a big set.  But, I think you're fine with `across` here.